bql/semantic/expression.go

```diff
@@ -90,11 +90,18 @@ type evaluationNode struct {
 }
 
 // format cell to a trimmed comparable string
-func formatCell(c *table.Cell) string {
+func formatCell(c *table.Cell) (string, error) {
 	if c.L != nil {
-		return strings.TrimSpace(c.L.ToComparableString())
+		return strings.TrimSpace(c.L.ToComparableString()), nil
 	}
-	return strings.TrimSpace(c.String())
+	if c.S != nil {
+		formatted, err := literal.DefaultBuilder().Build(literal.Text, *c.S)
+		if err != nil {
+			return "", fmt.Errorf("could not build literal from string, received error: %v", err)
```

Comment:
I wonder if we should annotate the error instead

Reply:
As discussed, `func Annotate(err error, msg string) error` not available out of Google3, proceeding with `fmt.Errorf` instead.